pkg/standalone/runconfig/run_config_parser_test.go

```diff
@@ -0,0 +1,141 @@
+
+	assert.Equal(t, 1, appsRunConfig.Version)
+	assert.NotEmpty(t, appsRunConfig.Common.ResourcesPath)
+	assert.NotEmpty(t, appsRunConfig.Common.Env)
+
+	firstAppConfig := appsRunConfig.Apps[0]
+	secondAppConfig := appsRunConfig.Apps[1]
+	assert.Equal(t, "", firstAppConfig.AppID)
+	assert.Equal(t, "GRPC", secondAppConfig.AppProtocol)
+	assert.Equal(t, 8080, firstAppConfig.AppPort)
+	assert.Equal(t, "", firstAppConfig.UnixDomainSocket)
+}
+
+func TestValidationsInRunConfig(t *testing.T) {
+	// tear down the created files.
+	tearDownCreatedFiles(t)
+
+	config := AppsRunConfig{}
+	config.ParseAppsConfig(configFilePath)
+
+	// check mangatory fields are not empty.
+	for _, app := range config.Apps {
+		assert.NotEmpty(t, app.AppDir)
+	}
+
+	// provided files/directories does not exist.
+	err := config.ValidateRunConfig()
+	assert.NotNil(t, err)
+
+	// create the files/directories provided in the config file.
+	createProvidedFiles(t, config)
+
+	// positive case- all files and directories exist.
+	err = config.ValidateRunConfig()
+	assert.Nil(t, err)
+
+	// negative case- app-dir field is empty.
+	temp := config.Apps[0].AppDir
+	config.Apps[0].AppDir = ""
+	err = config.ValidateRunConfig()
+	assert.NotNil(t, err)
+	config.Apps[0].AppDir = temp
+
+	// tear down the created files.
+	tearDownCreatedFiles(t)
+}
+
+func TestGetApps(t *testing.T) {
+	keymapping := []map[string]string{}
+	keymapping = append(
+		keymapping,
+		map[string]string{"AppHealthTimeout": "int", "app_dir": "string", "app_port": "int", "command": "[]interface {}", "config_file": "string", "resources_dir": "string"},
+		map[string]string{"app_dir": "string", "app_id": "string", "app_port": "int", "app_protocol": "string", "command": "[]interface {}", "env": "[]interface {}", "unix_domain_socket": "string"},
+	)
+
+	tearDownCreatedFiles(t)
+
+	config := AppsRunConfig{}
+	config.ParseAppsConfig(configFilePath)
+
+	// create the files/directories provided in the config file.
+	createProvidedFiles(t, config)
+
+	apps := config.GetApps(keymapping)
+
+	assert.Equal(t, 2, len(apps))
+	assert.Equal(t, "webapp", apps[0].AppID)
+	assert.Equal(t, "backend", apps[1].AppID)
+	assert.Equal(t, "HTTP", apps[0].AppProtocol)
+	assert.Equal(t, "GRPC", apps[1].AppProtocol)
+	assert.Equal(t, 8080, apps[0].AppPort)
+	assert.Equal(t, 1, apps[0].AppHealthTimeout)
+	assert.Equal(t, 10, apps[1].AppHealthTimeout)
+	assert.Equal(t, "", apps[0].UnixDomainSocket)
+	assert.Equal(t, "/tmp/test-socket", apps[1].UnixDomainSocket)
+
+	// tear down the created files.
+	tearDownCreatedFiles(t)
+}
+
+func createProvidedFiles(t *testing.T, config AppsRunConfig) {
+	err := os.MkdirAll(commonResourcesDir, os.ModePerm)
+	assert.Nil(t, err)
+	err = os.MkdirAll(app1ResourcesDir, os.ModePerm)
+	assert.Nil(t, err)
+	err = os.MkdirAll(app1Dir, os.ModePerm)
+	assert.Nil(t, err)
+	err = os.MkdirAll(app2Dir, os.ModePerm)
+	assert.Nil(t, err)
+	configFile, err := os.Create(app1ConfigFile)
+	assert.Nil(t, err)
+	configFile.Close()
+}
```

Comment:
Please create a temp directory and create all dirs and files under it

Reply:
@mukundansundar these file paths are same as in `test_app_config.yaml` file. So, whatever is mentioned there, same path I am creating here. Pinning to any particular directory like `tmp` would not help much I think. like for this we would have to specify the `tmp` directory in config file also which is diff across OS and also how will the actual `validate` method will know of this `tmp` directory.

Reply:
refactored it and made the complete path structure as part of the `testdata` folder.